packages/flutter_tools/lib/src/project.dart

```diff
@@ -1046,6 +1046,8 @@ class LinuxProject extends FlutterProjectPlatform {
   @override
   String get pluginConfigKey => LinuxPlugin.kConfigKey;
 
+  static final RegExp _applicationIdPattern = RegExp('^\\s*"application-id"\\s*,\\s*[\'"](.*)[\'"]\\s*,\\s*\$');
```

Comment:
Nit: using `r'...'` would mean you wouldn't need to do so much extra escaping; it's commonly used for defining regex strings.

Reply:
Actually, a second and more important change: you should make this a multi-line regex and allow for a newline between between the constant string and the ID (and remove the ^ from the start). If someone re-wrapped the file differently (e.g., running an auto-formatter with a long ID) this would break.

Right now it wouldn't matter since this is only used in a test where no editing has happened, but since this isn't test-only code it should be made robust against that.